geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/metadata/TableBasedMetadata.scala

```diff
@@ -99,7 +99,7 @@ trait TableBasedMetadata[T] extends GeoMesaMetadata[T] with LazyLogging {
   protected def scanKeys(): CloseableIterator[(String, String)]
 
   // only synchronize if table doesn't exist - otherwise it's ready only and we can avoid synchronization
-  private val tableExists: MaybeSynchronized[Boolean] =
+  private var tableExists: MaybeSynchronized[Boolean] =
```

Comment:
to keep this thread safe, you can just make this an AtomicBoolean, and then call `.set` on it in reset cache and ensureTableExists:
```
def ensureTableExists(): Unit = if (tableExists.compareAndSet(false, true)) { createTable() }
```

Reply:
> to keep this thread safe, you can just make this an AtomicBoolean, and then call `.set` on it in reset cache and ensureTableExists:
> 
> ```
> def ensureTableExists(): Unit = if (tableExists.compareAndSet(false, true)) { createTable() }
> ```

Using AtomicBoolean  caused a new problem:   `==`  is not work for value and expected in class IsSynchronized --`equals` method also not work because AtomicBoolean  does not override equlas method Object. Do we need to define IsSynchronized[AtomicBoolean] ? This does not seem to be a good idea.

Reply:
you can get rid of the whole MaybeSynchronized class. that was an optimization we can't use now that we are setting it back from true to false. the original idea was that in the usual case, where the table exists, that val would be `true` when created and then never change, so we don't need to synchronize on it.
so you'd do something like:

```
private val tableExists = new AtomicBoolean(checkIfTableExists)
...
def ensureTableExists(): Unit = if (tableExists.compareAndSet(false, true)) { createTable() }
...
override def resetCache(): Unit= {
  tableExists.set(checkIfTableExists)
  metaDataCache.invalidateAll()
}
```